static/app/components/events/interfaces/threads/threadSelector/index.tsx

```diff
@@ -50,28 +47,76 @@ function getThreadLabel(
 
 function ThreadSelector({threads, event, exception, activeThread, onChange}: Props) {
   const organization = useOrganization({allowNull: true});
+  const [currentThread, setCurrentThread] = useState<Thread>(activeThread);
+  const [sortAttribute, setSortAttribute] = useState<SortAttribute>(SortAttribute.ID);
+  const [isSortAscending, setIsSortAscending] = useState<boolean>(true);
 
   const hasThreadStates = threads.some(thread =>
     defined(getMappedThreadState(thread.state))
   );
+  const threadInfoMap = useMemo(() => {
+    return threads.reduce((acc, thread) => {
+      acc[thread.id] = filterThreadInfo(event, thread, exception);
+      return acc;
+    }, {});
+  }, [threads, event, exception]);
 
-  const items = useMemo(() => {
-    return threads.map((thread: Thread) => {
-      const threadInfo = filterThreadInfo(event, thread, exception);
-      return {
-        value: thread.id,
-        textValue: `#${thread.id}: ${thread.name} ${threadInfo.label} ${threadInfo.filename}`,
-        label: (
-          <Option
-            thread={thread}
-            details={threadInfo}
-            crashedInfo={threadInfo.crashedInfo}
-            hasThreadStates={hasThreadStates}
-          />
-        ),
-      };
+  const orderedThreads = useMemo(() => {
+    const sortedThreads = threads.sort((threadA, threadB) => {
```

Comment:
probably treat it as readonly
```suggestion
    const sortedThreads = threads.toSorted((threadA, threadB) => {
```